src/query/datavalues/src/columns/array/mod.rs

```diff
@@ -142,12 +143,15 @@ impl Column for ArrayColumn {
         if let ArrowType::LargeList(ref f) = arrow_type {
             let inner_f = from_arrow_field(f.as_ref());
             let array = self.values.as_arrow_array(inner_f);
-            Box::new(LargeListArray::from_data(
-                arrow_type,
-                self.offsets.clone(),
-                array,
-                None,
-            ))
+            Box::new(
+                LargeListArray::try_new(
+                    arrow_type,
+                    unsafe { OffsetsBuffer::new_unchecked(self.offsets.clone()) },
```

Comment:
We can use OffsetsBuffer::try_from

Reply:
I do think we should check the offset which drop the performance, we ensure it's valid.